open-api/rest-catalog-open-api.yaml

```diff
@@ -212,6 +212,34 @@ paths:
           schema:
             type: string
           example: "accounting%1Ftax"
+        - name: pageToken
+          in: query
+          description:
+            Allows clients to make use of pagination by signaling to the service that they would
+            prefer requests to be paginated to a “reasonable” number of results.
+            
+            New Clients always start the request by sending a required empty “pageToken” e.g. GET /tables?continuationToken=””
+            Servers supporting pagination would recognize pagination is requested due to the presence of the pageToken
+            and honor the contracts specified above by returning a non-empty continuation token
+            if there are more results available, or an empty token or missing field indicating no more results.
+            Servers that are non-pagination aware will ignore the token and return all results as they previously did.
+            
+            Old clients should not be specifying the new parameters
+            For servers that support pagination, they would notice the lack of “continuationToken”
```

Comment:
yeah this is the point we disagree. I think even when lacking a `pageToken` query parameter, the service should be able to paginate. Thoughts?

Reply:
So suppose we don't have empty page token, and service chooses what to do, either return full results or paginate. This means the old Iceberg REST clients will return incomplete results, seems like that is the issue.

On the other side, if we choose the token approach, we will also need a feature flag in the REST integrations, to use pagination or not. If the feature is turned off, then some REST services might run out of time before returning all contents, and have to fail the request. If the feature is turned on, then both REST services that support pagination or not can return full results.

Yes in that case I agree your approach is definitely more backwards compatible, but it is also cumbersome to even add a feature flag just for pagination. Let's see what others think.

Reply:
> yeah this is the point we disagree. I think even when lacking a pageToken query parameter, the service should be able to paginate. Thoughts?

Yea i think the issue is that without the `pageToken` how does the server tell if this is a new client that wants pagination or an old client that does not support it. 

 > Yes in that case I agree your approach is definitely more backwards compatible, but it is also cumbersome to even add a feature flag just for pagination. Let's see what others think.

Sounds good I think we have flags for other REST related features but not sure its too much to have a flag for pagination, will follow guidance from community on which path allows us to move forward. 
cc @danielcweeks @rdblue

Reply:
Wanted to further expand on this with a table shows the cases in how the `pageToken` will work for both old clients, new clients, old services and new services.
cc @jackye1995 @danielcweeks @rdblue 

|                     | Old services which does not support pagination | New Service which does not support pagination | New Service which only supports pagination | New Service (which supports both pagination and non pagination) |
| ----------- | ----------- |--------| ------- | -------- |
|Old Client (which does not send pageToken)  | current behavior return full results      |  current behavior return full result        | Will be expecting pageToken, and therefore will have to throw Unsupported Operation. In this case expectation is that client will have to upgrade to use pageToken.   | Either return all results or if taking too long throw Unsupported(saying must use pagination and upgrade to use the new client to send pageToken) |
|New Client (which sends pageToken) | would ingore pageToken and  default to current behavior return full result   |would ignore pageToken and  default to current behavior return full result   | return results in paginated way      | return results in paginated way  |